src/agent/h1-proxy-agent.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable max-classes-per-file */
+import { URL } from 'url';
+import net from 'net';
+import tls, { ConnectionOptions } from 'tls';
+import http, { ClientRequest, ClientRequestArgs } from 'http';
+import https from 'https';
+
+interface AgentOptions extends http.AgentOptions {
+    proxy: string | URL;
+}
+
+// const connectMap = new WeakMap<ClientRequest, true>();
```

Comment:
is this a WIP or should it be removed?

Reply:
Both. There is a `todo` 100 lines below to respect the `Agent` socket count limits. I initially thought this could be done by replacing `createSocket` or `createConnection` but this would need more investigation. It just works for now, shouldn't break.